res/controllers/engine-api.d.ts

```diff
@@ -31,6 +31,24 @@ declare interface ScriptConnection {
     readonly isConnected: boolean;
 }
 
+/** QmlPlayerProxy */
+declare interface PlayerInfo {
+    isLoaded(): boolean;
+    track(): string;
+    title(): string;
+    artist(): string;
+    album(): string;
+    albumArtist(): string;
+    genre(): string;
+    composer(): string;
+    grouping(): string;
+    year(): string;
+    trackNumber(): string;
+    trackTotal(): string;
+    comment(): string;
+    keyText(): string;
```

Comment:
Please add unit tests that:
-Test each function
-Ensure that the test case fails if somebody add a field to PlayerInfo. This is to ensure that nobody expose unintentional objects without implementing test and documentation

Reply:
Thats quite a lot of work and repeated code for not a lot of gain IMO...